builtin/logical/mysql/secret_creds.go

```diff
@@ -90,23 +90,31 @@ func (b *backend) secretCredsRevoke(
 	if !ok {
 		roleName = ""
 	} else {
-		roleName, ok = roleNameRaw.(string)
+		roleName, _ = roleNameRaw.(string)
 	}
 	// init default revoke sql string.
 	// this will replaced by a user provided one if one exists
 	// otherwise this is what will be used when lease is revoked
 	revokeSQL := defaultRevokeSQL
 
+	// init bool to track if we should responding with warning about nil role
+	nonNilResponse := false
+
 	// if we were successful in finding a role name
 	// create role entry from that name
 	if roleName != "" {
 		role, err := b.Role(req.Storage, roleName)
 		if err != nil {
 			return nil, err
 		}
+
+		if role == nil {
+			nonNilResponse = true
+		}
+
 		// Check for a revokeSQL string
 		// if one exists use that instead of the default
-		if role.RevokeSQL != "" {
+		if role.RevokeSQL != "" && role != nil {
```

Comment:
If role is nil this will panic. The if conditions are evaluated left to right so the check for role not being nil needs to come first.